src/mame/sony/news_68k_iop.cpp

```diff
@@ -0,0 +1,1150 @@
+        // Memory devices and main CPU's MMU
+        required_device<news_020_mmu_device> m_mmu;
+        required_device<ram_device> m_ram;
+        required_region_ptr<u32> m_eprom;
+        required_region_ptr<u32> m_idrom;
+
+        // Platform hardware
+        required_device<msm58321_device> m_rtc;
+        required_device<pit8253_device> m_interval_timer;
+        required_device<scc8530_device> m_scc_external;
+        required_device<scc8530_device> m_scc_peripheral;
+        required_device<am7990_device> m_net;
+        required_device<upd765a_device> m_fdc;
+        required_device<ncr5380_device> m_scsi;
+        required_device<news_iop_scsi_helper_device> m_scsi_dma;
+        required_ioport m_dip_switch;
+        required_device_array<rs232_port_device, 2> m_serial;
+        std::unique_ptr<u16[]> m_net_ram;
+
+        // IOP IRQ state
+        uint32_t m_iop_intst = 0;
+        uint32_t m_iop_inten = 0;
+        std::map<int, bool> m_iop_int_state = {
+            {INPUT_LINE_IRQ1, false},
+            {INPUT_LINE_IRQ2, false},
+            {INPUT_LINE_IRQ3, false},
+            {INPUT_LINE_IRQ4, false},
+            {INPUT_LINE_IRQ5, false},
+            {INPUT_LINE_IRQ6, false},
+            {INPUT_LINE_IRQ7, false}
+        };
+
+        // CPU IRQ state
+        uint32_t m_cpu_intst = 0;
+        uint32_t m_cpu_inten = 0;
+        std::map<int, bool> m_cpu_int_state = {
+            {INPUT_LINE_IRQ1, false},
+            {INPUT_LINE_IRQ2, false},
+            {INPUT_LINE_IRQ3, false},
+            {INPUT_LINE_IRQ4, false},
+            {INPUT_LINE_IRQ5, false},
+            {INPUT_LINE_IRQ6, false},
+            {INPUT_LINE_IRQ7, false}
+        };
+
+        // Bus error details
+        bool m_cpu_bus_error = false;
+        offs_t m_last_berr_pc_cpu = 0;
+        uint8_t m_cpu_bus_error_status = 0;
+        emu_timer *m_cpu_bus_error_timer;
+
+        // Front panel
+        bool m_panel_clear = false;
+        bool m_panel_shift = false;
+        int m_sw1_first_read = 0;
+        int m_panel_shift_count = 0;
+        uint8_t m_rtc_data = 0;
+
+        // Other constants
+        static constexpr int NET_RAM_SIZE = 8192; // 16K RAM, in 16-bit words
+    };
+
+    const std::map<int, std::vector<news_iop_state::iop_irq_number>> news_iop_state::iop_irq_line_map = {
+        {INPUT_LINE_IRQ1, {CPIRQ_3_1}},
+        {INPUT_LINE_IRQ2, {SCSI_IRQ, SCSI_DRQ}},
+        {INPUT_LINE_IRQ3, {LANCE}},
+        {INPUT_LINE_IRQ4, {CPU}},
+        {INPUT_LINE_IRQ5, {SCC, SCC_PERIPHERAL}},
+        {INPUT_LINE_IRQ6, {TIMEOUT, FDCIRQ}},
+        {INPUT_LINE_IRQ7, {FDCDRQ}}
+    };
+
+    const std::map<int, news_iop_state::cpu_irq_number> news_iop_state::cpu_irq_line_map = {
+        // AST is excluded from this check
+        {INPUT_LINE_IRQ2, CPIRQ1},
+        {INPUT_LINE_IRQ3, IOPIRQ3},
+        {INPUT_LINE_IRQ4, CPIRQ3},
+        {INPUT_LINE_IRQ5, IOPIRQ5},
+        {INPUT_LINE_IRQ6, TIMER},
+        {INPUT_LINE_IRQ7, PERR}
+    };
```

Comment:
Please don’t use static variables that require runtime allocation.  It slows down startup for *all* situations.

Reply:
Changed to initialization in the ctor instead

Reply:
As an optimisation, you might be able to get away with turning them into some kind of static `std::array` variables and then using `std::lower_bound` to find things.